### PR TITLE
Reject duplicate AccessToken names

### PR DESCRIPTION
**Pull request by 6543**
as title
source: https://github.com/go-gitea/gitea/pull/7135

### CODE REVIEW

**Comment by 6543** [comment]
Ready for Review

File: models/token.go [hunk 1]
```diff
@@ -77,6 +77,12 @@ func GetAccessTokenBySHA(token string) (*AccessToken, error) {
 	return nil, ErrAccessTokenNotExist{token}
 }
 
+// AccessTokenByNameExists checks if a token name has been used already by a user.
+func AccessTokenByNameExists(token *AccessToken) bool {
+	exists, _ := x.Table("access_token").Where("name = ?", token.Name).And("uid = ?", token.UID).Exist()
```
Comment (lafriks) [inline models/token.go]: Error should not be ignored and returned down to caller where status code 500 should be returned in case of error

**Comment by 6543** [comment]
@lafriks done

**Review by lafriks — APPROVED** [review]

**Comment by 6543** [comment]
UPDATE: need some fixes for tests

**Comment by 6543** [comment]
@lafriks hope you are fine with 50e8f9235dc603955c567545e6bc1b254ee012c8
since Tests dont runn in paralell there should be no race

**Comment by lafriks** [comment]
Make lgtm work

**Comment by 6543** [comment]
@lafriks sadly there is only one review :(

**Comment by lafriks** [comment]
Yeah, only later noticed :)

**Review by zeripath — APPROVED** [review]

**Comment by zeripath** [comment]
Having thought long about this, I think it's probably reasonable to disallow duplicate names - because it could be confusing for users. Therefor LGTM.

**Comment by lafriks** [comment]
make lgtm to me again ;)

**Pull request merged**